include/toml++/impl/path.h

```diff
@@ -0,0 +1,438 @@
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& append(std::wstring_view);
+
+		/// \brief	Prepends elements to the beginning of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& prepend(std::wstring_view);
+
+		/// \brief	Replaces the contents of the path object by a new path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& assign(std::wstring_view);
+
+		/// \brief Returns a wstring representing the path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		std::wstring wstring() const;
+
+#endif // TOML_ENABLE_WINDOWS_COMPAT
+
+		// === Hidden friend operators ========================================
+		friend std::ostream& operator<<(std::ostream& os, const toml::path& rhs)
+		{
+			os << rhs.string();
+			return os;
+		}
+
+		friend std::istream& operator>>(std::istream& is, toml::path& rhs)
+		{
+			std::string s;
+			is >> s;
+			rhs.assign(s);
+
+			return is;
+		}
+
+		friend path operator+(const toml::path& lhs, std::string_view rhs)
+		{
+			return lhs + toml::path(rhs);
+		}
+
+		friend path operator+(const toml::path& lhs, const char* rhs)
+		{
+			return lhs + toml::path(rhs);
+		}
+
+		friend path operator+(std::string_view lhs, const toml::path& rhs)
+		{
+			toml::path result = { rhs };
+			result.prepend(lhs);
+
+			return result;
+		}
+
+		friend path operator+(const char* lhs, const toml::path& rhs)
+		{
+			toml::path result = { rhs };
+			result.prepend(lhs);
+
+			return result;
+		}
+
+		friend path operator/(const toml::path& lhs, std::string_view rhs)
+		{
+			return lhs + rhs;
+		}
+
+		friend path operator/(const toml::path& lhs, const char* rhs)
+		{
+			return lhs + toml::path(rhs);
+		}
+
+		friend path operator/(std::string_view lhs, const toml::path& rhs)
+		{
+			return lhs + rhs;
+		}
+
+		friend path operator/(const char* lhs, const toml::path& rhs)
+		{
+			return lhs + rhs;
+		}
+
+#if TOML_ENABLE_WINDOWS_COMPAT
+
+		friend path operator+(const toml::path& lhs, std::wstring_view rhs)
+		{
+			return lhs + toml::path(rhs);
+		}
+
+		friend path operator+(const toml::path& lhs, const wchar_t* rhs)
+		{
+			return lhs + toml::path(rhs);
+		}
+
+		friend path operator+(std::wstring_view lhs, const toml::path& rhs)
+		{
+			toml::path result = { rhs };
+			result.prepend(lhs);
+
+			return result;
+		}
+
+		friend path operator+(const wchar_t* lhs, const toml::path& rhs)
+		{
+			toml::path result = { rhs };
+			result.prepend(lhs);
+
+			return result;
+		}
+
+		friend path operator/(const toml::path& lhs, std::wstring_view rhs)
+		{
+			return lhs + rhs;
+		}
+
+		friend path operator/(const toml::path& lhs, const wchar_t* rhs)
+		{
+			return lhs + toml::path(rhs);
+		}
+
+		friend path operator/(std::wstring_view lhs, const toml::path& rhs)
+		{
+			return lhs + rhs;
+		}
+
+		friend path operator/(const wchar_t* lhs, const toml::path& rhs)
+		{
+			return lhs + rhs;
+		}
+
+#endif // TOML_ENABLE_WINDOWS_COMPAT
+
+	};
+
+	inline namespace literals
+	{
+		TOML_ABI_NAMESPACE_BOOL(TOML_EXCEPTIONS, lit_ex, lit_noex);
```

Comment:
```suggestion
```

You don't need the inner ABI namespace here. That was specifically relating to whether or not the `toml::parse()` can throw, based on the setting for `TOML_EXCEPTIONS` - the ABI namespace in that scenario makes it so that people won't have weird errors if they accidently compile the library in one mode, and their application in another - instead it forces a hard linker error.

(again, something I've left undocumented and probably should note somewhere)